### PR TITLE
fulltext search: add pipeline and fulltext indexer

### PR DESCRIPTION
**Pull request by MJedr**
ref:#2436

### CODE REVIEW

File: backend/inspirehep/indexer/cli.py [hunk 1]
```diff
@@ -163,7 +165,10 @@ def reindex_records(
         record = InspireRecord.get_record_by_pid_value(
             pid_value, pid_type, original_record=True
         )
-        record.index()
+        if pid_type == "lit" and fulltext:
+            record.index_fulltex()
```
Comment (drjova) [inline backend/inspirehep/indexer/cli.py:169]:
```suggestion
            record.index_fulltext()
```

File: backend/inspirehep/indexer/base.py [hunk 2]
```diff
@@ -132,3 +134,62 @@ def bulk_action(self, record_uuid):
             LOGGER.exception(
                 "Kombu is not able to process response!", uuid=str(record_uuid)
             )
+
+
+class InspireRecordFulltextIndexer(InspireRecordIndexer):
```
Comment (drjova) [inline backend/inspirehep/indexer/base.py:139]: isn't it `LiteratureRecordFulltextIndexer`?

Comment (MJedr) [reply backend/inspirehep/indexer/base.py:139]: yes, would be better probably

File: backend/inspirehep/records/receivers.py [hunk 3]
```diff
@@ -44,3 +45,8 @@ def index_after_commit(sender, changes):
                     author_disambiguation(model_instance)
                 if "new_record" in model_instance.json:
                     redirect_references_to_merged_record.delay(str(model_instance.id))
+                if (
+                    "documents" in model_instance.json
+                    and current_app.config["FEATURE_FLAG_ENABLE_FULLTEXT"]
+                ):
+                    index_fulltext.delay(str(model_instance.id))
```
Comment (drjova) [inline backend/inspirehep/records/receivers.py:52]: shouldn't we only run it for literature?

File: backend/tests/integration-async/records/indexer/test_literature.py [hunk 4]
```diff
@@ -466,3 +466,68 @@ def assert_record_is_deleted_from_es():
...
+        lit_record = LiteratureRecord.create(
+            faker.record(
+                "lit",
+                data={
+                    "documents": [
+                        {
+                            "source": "arxiv",
+                            "fulltext": True,
+                            "key": "new_doc.pdf",
+                            "url": "http://www.africau.edu/images/default/sample.pdf",
```
Comment (drjova) [inline backend/tests/integration-async/records/indexer/test_literature.py:484]: I would strongly recommend add a cassette for these tests

File: backend/inspirehep/records/receivers.py (hunk 3 above)
Comment (MJedr) [reply backend/inspirehep/records/receivers.py:52]: I did a check in a task for the record type but I can do it here (also don't think that we have `documents` in any other record)

Comment (drjova) [reply backend/inspirehep/records/receivers.py:52]: we will have in the `data` collection is coming on Q3

File: backend/inspirehep/indexer/tasks.py [hunk 5]
```diff
@@ -108,3 +111,34 @@ def index_record(self, uuid, record_version=None, force_delete=None):
...
+)
+def index_fulltext(self, record_id):
+    record = InspireRecord.get_record(record_id)
+    LOGGER.info("Indexing record including fulltext", uuid=record.id)
+    if record.pid_type == "lit":
+        LiteratureRecordFulltextIndexer().index_fulltext(record)
+
+
+@shared_task(ignore_result=False, bind=True)
+def batch_index_fulltext(self, records_uuids, request_timeout=None):
```
Comment (drjova) [inline backend/inspirehep/indexer/tasks.py:132]:
```suggestion
def batch_index_literature_fulltext(self, records_uuids, request_timeout=None):
```

**Review by drjova — COMMENTED** [review]
@MJedr LGTM to start testing, just the comment

**Review by drjova — APPROVED** [review]

**Pull request merged**